src/util.ts

```diff
@@ -0,0 +1,87 @@
+/**
+ * Based on environment, get a reference to the Web Crypto API's SubtleCrypto interface.
+ * @returns An implementation of the Web Crypto API's SubtleCrypto interface.
+ */
+function getSubtleCrypto(): SubtleCrypto {
+	if (typeof window !== 'undefined' && window.crypto) {
+		return window.crypto.subtle;
+	}
+	if (typeof globalThis !== 'undefined' && globalThis.crypto) {
+		return globalThis.crypto.subtle;
+	}
+	if (typeof crypto !== 'undefined') {
+		return crypto.subtle;
+	}
+	if (typeof require === 'function') {
+		// Cloudflare Workers are doing what appears to be a regex check to look and
+		// warn for this pattern. We should never get here in a Cloudflare Worker, so
+		// I am being coy to avoid detection and a warning.
+		const cryptoPackage = 'node:crypto';
+		const crypto = require(cryptoPackage);
+		return crypto.webcrypto.subtle;
+	}
+	throw new Error('No Web Crypto API implementation found');
+}
+
+export const subtleCrypto = getSubtleCrypto();
+
+/**
+ * Converts different types to Uint8Array.
+ *
+ * @param value - Value to convert. Strings are parsed as hex.
+ * @param format - Format of value. Valid options: 'hex'. Defaults to utf-8.
+ * @returns Value in Uint8Array form.
+ */
+export function valueToUint8Array(
+	value: Uint8Array | ArrayBuffer | Buffer | string,
+	format?: string,
+): Uint8Array {
+	if (value == null) {
+		return new Uint8Array();
+	}
+	if (typeof value === 'string') {
+		if (format === 'hex') {
+			const matches = value.match(/.{1,2}/g);
+			if (matches == null) {
+				throw new Error('Value is not a valid hex string');
+			}
+			const hexVal = matches.map((byte: string) => Number.parseInt(byte, 16));
+			return new Uint8Array(hexVal);
+		}
+
+		return new TextEncoder().encode(value);
+	}
+	try {
+		if (Buffer.isBuffer(value)) {
+			return new Uint8Array(value);
+		}
+	} catch (ex) {
+		// Runtime doesn't have Buffer
```

Comment:
Should we do a `typeof Buffer !== 'undefined'` check here in case we're swallowing exceptions?

Reply:
I can't imagine what other kind of exception could happen in this very narrow use case